nservicebus/pipeline/manipulate-with-behaviors_registernew_core_[6.0,).partial.md

```diff
@@ -5,4 +5,6 @@ snippet:RegisterBehaviorEndpointConfiguration
 
 Behaviors can also be registered from a `Feature` as shown below:
 
-snippet:RegisterBehaviorFromFeature
\ No newline at end of file
+snippet:RegisterBehaviorFromFeature
+
+WARNING: Behaviors are only created once and the same instance is reused on every invocation of the pipeline, even when registering the behavior to be created by a Dependency Injection Container. Never use dependency injection for dependencies configured with an *InstancePerUnitOfWork* scope.
```

Comment:
Would it be better to say:

Behaviors MUST only depend on components configured as singletons (link to DI doco)  since they are created once .....

?